baseplate/server.go

```diff
@@ -0,0 +1,70 @@
+package baseplate
+
+import (
+	"errors"
+	"io/ioutil"
+	"time"
+
+	"github.com/reddit/baseplate.go/log"
+	"gopkg.in/yaml.v2"
+)
+
+// ServerConfig is a general purpose config for assembling a BaseplateServer
+type ServerConfig struct {
+	Addr string
+
+	Timeout time.Duration
+
+	Log struct {
+		Level log.Level
+	}
+
+	Metrics struct {
+		Namespace string
+		Endpoint  string
+	}
+
+	Secrets struct {
+		Path string
+	}
+
+	Sentry struct {
+		DSN         string
+		Environment string
+		SampleRate  float64
+	}
+
+	Tracing struct {
+		Namespace     string
+		Endpoint      string
+		RecordTimeout time.Duration `yaml:"recordTimeout"`
```

Comment:
Why is this the only one in the whole `ServerConfig` with yaml annotation?

Reply:
lift-n-shift